packages/web-frontend/src/containers/Map/AreaTooltip.js

```diff
@@ -34,8 +41,76 @@ export class AreaTooltip extends Component {
     }
   }
 
+  getTextList() {
+    const { orgUnitName, hasMeasureValue, measureOptions, orgUnitMeasureData } = this.props;
+
+    const defaultTextList = [
+      <Heading key={0} hasMeasureValue={hasMeasureValue}>
+        {orgUnitName}
+      </Heading>,
+    ];
+
+    const renderTextList = (data, textList = []) => {
+      Object.keys(data).forEach(key => {
+        textList.push(<span key={key}>{`${key}: ${data[key]}`}</span>);
+      });
+      return textList;
+    };
+
+    if (hasMeasureValue) {
+      const legacy = measureOptions.length === 1;
+      const { formattedMeasureData, onlyOneValue } = legacy
+        ? this.buildLegacyFormattedMeasureData(orgUnitMeasureData, measureOptions)
+        : this.buildFormattedMeasureData(orgUnitMeasureData, measureOptions);
+
+      if (onlyOneValue) {
+        return renderTextList({ [orgUnitName]: formattedMeasureData });
+      }
+
+      renderTextList(formattedMeasureData, defaultTextList);
+    }
+
+    return defaultTextList;
+  }
+
+  buildLegacyFormattedMeasureData(orgUnitMeasureData, measureOptions) {
+    const formattedValue = getSingleFormattedValue(orgUnitMeasureData, measureOptions);
+
+    return {
+      formattedMeasureData: formattedValue,
+      onlyOneValue: true,
+    };
+  }
+
+  buildFormattedMeasureData(orgUnitMeasureData, measureOptions) {
+    const formattedMeasureData = {};
+
+    if (orgUnitMeasureData) {
+      formattedMeasureData[measureOptions[0].name] = getSingleFormattedValue(
+        orgUnitMeasureData,
+        measureOptions,
+      );
+
+      measureOptions.forEach(({ key }) => {
+        if (key !== measureOptions[0].key) {
+          formattedMeasureData[key] = orgUnitMeasureData[key];
+        }
+      });
+    }
```

Comment:
Is there a reason we're not treating the first measure option and all the others the same? Those other options might also have `valueType` fields or something like that right? Could this just be:
```suggestion
      measureOptions.forEach(({ key, name, valueType }) => {
        const metadata = data[`${dataKey}_metadata`] || data[`${data.name}_metadata`] || {};
        formattedMeasureData[name || key] = formatDataValueByType({ value: orgUnitMeasureData[key], metadata }, valueType);
      });
    }
```

Reply:
I was thinking it can leave formating work to `report-server` so that we don't need to increase frontend loading time?

Reply:
That's a good concern to have, but I don't think it would affect performance at all, and the consistency between measure options will be very nice 👍

Reply:
New changes will let report-server support using metadata only with this formate `${key}_metadata`:
```
key1: 0.02,
key1_metadata_numerator: 2,
key1_metadata_denominator: 100,
key2: 0.01,
key2_metadata_numerator: 1,
key2_metadata_denominator: 100,
```